client/sites-dashboard/components/sites-list-sorting-dropdown.tsx

```diff
@@ -0,0 +1,125 @@
+import { Gridicon, SitesTableSortKey, SitesTableSortOrder } from '@automattic/components';
+import styled from '@emotion/styled';
+import { Button, Dropdown, MenuGroup, MenuItem } from '@wordpress/components';
+import { useI18n } from '@wordpress/react-i18n';
+import { useMemo } from 'react';
+import { useAsyncPreference } from 'calypso/state/preferences/use-async-preference';
+
+const SortingButton = styled( Button )( {
+	alignSelf: 'stretch',
+	flexDirection: 'row-reverse',
+	gap: '4px',
+	whiteSpace: 'nowrap',
+} );
+
+const SortingButtonIcon = styled( Gridicon )( {
+	marginRight: '0 !important',
+} );
+
+const SEPARATOR = '-' as const;
+
+type SitesListSorting = `${ SitesTableSortKey }${ typeof SEPARATOR }${ SitesTableSortOrder }`;
+
+const DEFAULT_LIST_SORTING = {
+	sortKey: 'updatedAt',
+	sortOrder: 'desc',
+} as const;
+
+export const parseSorting = ( sorting: SitesListSorting | 'none' ) => {
+	if ( sorting === 'none' ) {
+		return DEFAULT_LIST_SORTING;
+	}
+
+	const [ sortKey, sortOrder ] = sorting.split( SEPARATOR );
+
+	return {
+		sortKey: sortKey as SitesTableSortKey,
+		sortOrder: sortOrder as SitesTableSortOrder,
+	};
+};
+
+export const useSitesListSorting = () =>
+	useAsyncPreference< SitesListSorting >( {
+		defaultValue: `${ DEFAULT_LIST_SORTING.sortKey }-${ DEFAULT_LIST_SORTING.sortOrder }`,
+		preferenceName: 'sites-list-sorting',
+	} );
+
+interface SitesListSortingDropdownProps {
+	onSitesListSortingChange( newValue: SitesListSorting ): void;
+	sitesListSorting: ReturnType< typeof useSitesListSorting >[ 0 ];
+}
+
+export const SitesListSortingDropdown = ( {
+	onSitesListSortingChange,
+	sitesListSorting,
+}: SitesListSortingDropdownProps ) => {
+	const { __ } = useI18n();
+
+	const label = useMemo( () => {
+		if ( sitesListSorting === 'none' ) {
+			return null;
+		}
+
+		switch ( sitesListSorting ) {
+			case `alphabetically${ SEPARATOR }asc`:
+				return __( 'Sorted alphabetically (A-Z)' );
+
+			case `alphabetically${ SEPARATOR }desc`:
+				return __( 'Sorted alphabetically (Z-A)' );
+
+			case `updatedAt${ SEPARATOR }desc`:
+				return __( 'Sorted by last publish date' );
+
+			default:
+				throw new Error( `invalid sort value ${ sitesListSorting }` );
+		}
+	}, [ __, sitesListSorting ] );
+
+	if ( sitesListSorting === 'none' ) {
+		return null;
+	}
+
+	return (
+		<Dropdown
+			position="bottom center"
+			renderToggle={ ( { isOpen, onToggle } ) => (
+				<SortingButton
+					icon={ <SortingButtonIcon icon={ isOpen ? 'arrow-up' : 'arrow-down' } /> }
```

Comment:
To me, the arrow seems to indicate 'order by':

<img width="378" alt="image" src="https://user-images.githubusercontent.com/36432/189740782-4093cd79-4589-4f63-909f-ad06c9c04f8c.png">

What do you think about using a caret, like Slack?

<img width="254" alt="image" src="https://user-images.githubusercontent.com/36432/189740835-1587baeb-7be3-416e-97ef-f1b7fb235694.png">

Reply:
Makes sense, will give it a try.

Reply:
Much better. Closing this.